lib/logflare/logs/vercel/nimble_vercel_parser.ex

```diff
@@ -168,30 +171,28 @@ defmodule Logflare.Logs.Vercel.NimbleLambdaMessageParser do
     message = Keyword.get_values(tokens, :message) |> Enum.join("\n")
 
     if maybe_json do
-      maybe_json
-      |> String.replace("\n", "")
-      |> JSON.decode()
-      |> case do
-        {:ok, json} ->
-          %{"data" => json, "message" => message}
-
-        _ ->
-          result = maybe_parse_multiline_json_body(maybe_json)
-
-          case result do
-            {"lines", lines} ->
-              %{"multiline" => lines}
-
-            _ ->
-              message = Keyword.values(tokens) |> Enum.join("")
-              %{"message" => message}
-          end
-      end
+      build_map(maybe_json, message, tokens)
     else
       %{"message" => message}
     end
   end
 
+  defp build_map(maybe_json, message, tokens) do
+    cleaned_json = String.replace(maybe_json, "\n", "")
+
+    with {:error, _} <- JSON.decode(cleaned_json),
+         {"lines", lines} <- maybe_parse_multiline_json_body(maybe_json) do
+      %{"multiline" => lines}
+    else
+      {:ok, json} ->
+        %{"data" => json, "message" => message}
```

Comment:
would prefer the happy path to be within the with/do 🙏

Reply:
Will change it :pray: